### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by uuf6429**
Based on #275.

Mainly this is a small update to documentation (contrib and readme).
There are, however, some additional non-functional fixes:
- added a basic .editorconfig
- updated gitattributes
- gitignored local phpunit.xml file (standard convention)
- updated documentation:
  - used hashes instead of lines for headers
  - fixed some wording
  - used proper code fencing (no space before language)
  - `shell` instead of `bash`
  - fixed broken reference to CHANGES.md
  - http -> https
- initial work for linting tools* (empty composer script; section in documentation)

\* which will come soon, hopefully.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
-/package.xml.tpl export-ignore
-/phpdoc.ini.dist export-ignore
-/phpunit.xml.dist export-ignore
+/.github/           export-ignore
+/bin/               export-ignore
+/tests              export-ignore
+/.editorconfig      export-ignore
+/.gitattributes     export-ignore
+/.gitignore         export-ignore
+/CHANGELOG.md       export-ignore
```
Comment (uuf6429) [inline .gitattributes:7]: I'm wondering if it actually might make sense to allow CHANGELOG.md in the export?

File: .gitattributes [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
+/CONTRIBUTING.md    export-ignore
+/package.xml.tpl    export-ignore
+/phpdoc.ini.dist    export-ignore
+/phpunit.dist.xml   export-ignore
```
Comment (uuf6429) [inline .gitattributes:11]: Hmm, I just realised .dist.xml is only supported since PHPUnit 10. :(

Comment (uuf6429) [reply .gitattributes:11]: Hmm, I'll move this part to a different PR... will be less things to argue about.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -14,31 +14,26 @@ jobs:
...
-          if: matrix.symfony-version != ''
-          run: composer require --no-update "symfony/symfony:${{ matrix.symfony-version }}"
-
-      -   name: Install dependencies
-          run: composer update ${{ matrix.composer-flags }}
-
-      -   name: Run tests (phpunit)
-          run: ./vendor/bin/phpunit
+      - uses: actions/checkout@v4
+      - name: Setup PHP
```
Comment (acoulton) [inline .github/workflows/build.yml:27]: Personally, I find it a lot easier to read / see / review YAML with a blank line between each block as before

File: .gitattributes (hunk 1 above)
Comment (acoulton) [reply .gitattributes:7]: Yeah, I think I would keep the CHANGELOG and the README in the distribution package

File: .phpunit.cache/test-results [hunk 4]
```diff
@@ -0,0 +1 @@
...hat\\Gherkin\\ParserTest::testParser#outline_with_multiple_examples":0.002,"Tests\\Behat\\Gherkin\\ParserTest::testParser#outline_with_spaces":0.002,"Tests\\Behat\\Gherkin\\ParserTest::testParser#outline_with_step_table":0.002,"Tests\\Behat\\Gherkin\\ParserTest::testParser#pystring":0.001,"Tests\\Behat\\Gherkin\\ParserTest::testParser#ru_addition":0.002,"Tests\\Behat\\Gherkin\\ParserTest::testParser#ru_commented":0.001,"Tests\\Behat\\Gherkin\\ParserTest::testParser#ru_consecutive_calculations":0.002,"Tests\\Behat\\Gherkin\\ParserTest::testParser#ru_division":0.002,"Tests\\Behat\\Gherkin\\ParserTest::testParser#start_comments":0.001,"Tests\\Behat\\Gherkin\\ParserTest::testParser#tables":0.002,"Tests\\Behat\\Gherkin\\ParserTest::testParser#tags_sample":0.004,"Tests\\Behat\\Gherkin\\ParserTest::testParser#test_unit":0.001,"Tests\\Behat\\Gherkin\\ParserTest::testParser#trimpystring":0.001,"Tests\\Behat\\Gherkin\\ParserTest::testParser#undefined_multiline_args":0.001,"Tests\\Behat\\Gherkin\\ParserTest::testParserResetsTagsBetweenFeatures":0,"Tests\\Behat\\Gherkin\\ParserTest::testSingleCharacterStepSupport":0,"Tests\\Behat\\Gherkin\\ParserTest::testParsingManyCommentsShouldPass":0.001}}
```
Comment (acoulton) [inline .phpunit.cache/test-results:1]: Should this have been committed?

File: README.md [hunk 5]
```diff
@@ -1,20 +1,17 @@
...
 
 This is the php Gherkin parser for Behat. It comes bundled with more than 40 native languages
 (see `i18n.php`) support & clean architecture.
 
-Useful Links
-------------
+## Useful Links
 
 - Official Google Group is at [http://groups.google.com/group/behat](http://groups.google.com/group/behat)
 - IRC channel on [#freenode](http://freenode.net/) is `#behat`
```
Comment (acoulton) [inline README.md:9]: We can remove these, I think. Maybe add a link to the main behat website?

File: package.xml.tpl [hunk 6]
```diff
@@ -38,7 +38,7 @@
             <file role="php" baseinstalldir="gherkin" name="vendor/.composer/autoload.php" />
             <file role="php" baseinstalldir="gherkin" name="vendor/.composer/autoload_namespaces.php" />
             <file role="php" baseinstalldir="gherkin" name="phpdoc.ini.dist" />
-            <file role="php" baseinstalldir="gherkin" name="phpunit.xml.dist" />
+            <file role="php" baseinstalldir="gherkin" name="phpunit.dist.xml" />
```
Comment (acoulton) [inline package.xml.tpl:41]: Is this `package.xml.tpl` actually used for anything? It looks quite outdated (minimum php = 5.3.1, for example).

File: phpunit.dist.xml [hunk 7]
```diff
@@ -1,25 +1,21 @@
...
-<phpunit backupGlobals="false"
-         backupStaticAttributes="false"
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/10.5/phpunit.xsd"
+         backupGlobals="false"
          colors="true"
-         convertDeprecationsToExceptions="true"
-         convertErrorsToExceptions="true"
-         convertNoticesToExceptions="true"
-         convertWarningsToExceptions="true"
```
Comment (acoulton) [inline phpunit.dist.xml:9]: I think we need to add the equivalent phpunit 10 failure modes e.g.
```
         failOnDeprecation="true"
         failOnNotice="true"
         failOnWarning="true"
         failOnRisky="true"
         displayDetailsOnIncompleteTests="true"
         displayDetailsOnSkippedTests="true"
         displayDetailsOnTestsThatTriggerDeprecations="true"
         displayDetailsOnTestsThatTriggerErrors="true"
         displayDetailsOnTestsThatTriggerNotices="true"
         displayDetailsOnTestsThatTriggerWarnings="true"
```

**Review by acoulton — COMMENTED** [review]
thanks, a few thoughts

**Comment by acoulton** [comment]
oops sorry, just seen you submitted #275 separately

**Review by acoulton — APPROVED** [review]
LGTM, thanks 😀

**Review by carlos-granados — APPROVED** [review]
Looks good, thanks @uuf6429 There are some conflicts that need to be solved, could you please look into that?

**Pull request merged**